hummingbot/core/gateway/gateway_http_client.py

```diff
@@ -492,7 +492,7 @@ async def amm_trade(
             "quote": quote_asset,
             "side": side.name,
             "amount": f"{amount:.18f}",
-            "limitPrice": str(price),
+            "limitPrice": f"{price:.20f}",
```

Comment:
This change is actually right in the bounty description, maybe we should add more value to this PR by adding test case to check for this issue?